packages/ngtools/webpack/src/paths-plugin.ts

```diff
@@ -154,40 +158,73 @@ export class TypeScriptPathsPlugin {
         // For example, if the first one resolves, any others are not needed and do not need
         // to be created.
         const replacements = findReplacements(originalRequest, this.patterns);
+        const basePath = this.baseUrl ?? '';
+
+        const attemptResolveRequest = (request: DoResolveValue): Promise<DoResolveValue | null> => {
+          return new Promise((resolve, reject) => {
+            resolver.doResolve(
+              target,
+              request,
+              '',
+              resolveContext,
+              (error: Error | null, result: DoResolveValue) => {
+                if (error) {
+                  reject(error);
+                } else if (result) {
+                  resolve(result);
+                } else {
+                  resolve(null);
+                }
+              },
+            );
+          });
+        };
 
-        const tryResolve = () => {
+        const tryNextReplacement = () => {
           const next = replacements.next();
           if (next.done) {
             callback();
 
             return;
           }
 
-          const potentialRequest = {
+          const targetPath = path.resolve(basePath, next.value);
+          const canBeModule = path.extname(targetPath) !== '';
```

Comment:
should this be
```suggestion
          const canBeModule = path.extname(targetPath) === '';
```

Reply:
As discussed offline, the tests worked because the TS file actually exist and just when package resolution would come first, Webpack would transpile the file and the `module-resolution` spec would pass unexpectedly. If file resolution comes first (like it does now), then Webpack will complain about the TS file being outside of the sources/compilation.